### ISSUE TITLE
Easy Means of Enabling Swap Memory on Podman Machine (MacOS, maybe others)

### ISSUE THREAD

**Issue by jamesmikesell** [containers/podman#15980]
/kind feature

**Description**

I'm wondering if there is an easy way to enable / set swap memory on the podman machine, similar to how memory can be set during the init by using the `-m` flag.

IE Being able to do something like this:
```shell
podman machine init -m 4096 --swap 6000
```

Is there a reason swap is disabled on the underlying VM by default?

**Comment by mheon** [comment]
@ashley-cui PTAL

**Comment by afbjorklund** [comment]
Something like this (swap file) could work, as a workaround:

https://github.com/coreos/docs/blob/master/os/adding-swap.md

**Comment by rhatdan** [comment]
@ashley-cui Any movement on this?
@afbjorklund @jamesmikesell interested in opening a PR?

**Comment by ashley-cui** [comment]
No movement at the moment, going to unassign for now and pick it up later if no one else has

**Comment by jamesmikesell** [comment]
@afbjorklund  i had something similar in my notes on how to add swap by ssh-ing into the vm... however something simpler would certainly be welcome.

Here are my notes:

```shell
podman machine ssh

sudo dd if=/dev/zero of=/var/swap-file bs=1024 count=6000000
sudo chmod 600 /var/swap-file
sudo mkswap /var/swap-file
sudo swapon /var/swap-file
echo "/var/swap-file none swap sw 0 0" | sudo tee -a /etc/fstab
```

**Comment by rhatdan** [comment]
@benoitf Is this something we should add to Podman-Desktop?

**Comment by benoitf** [comment]
@rhatdan Podman Desktop could call/mnage the command but IMHO flag should be available through podman machine cli as well.

**Comment by rhatdan** [comment]
@baude @ashley-cui @n1hility WDYT?

**Comment by n1hility** [comment]
I agree that it makes sense to provide this for  mac & linux. On Windows things are different since they are tuned automatically and are global to all other running WSL distros and podman machine instances. IMO best would be to treat those values on Win as a noop like we do with the m flag.

**Comment by rhatdan** [comment]
@jamesmikesell Interested in opening a PR for this?

**Comment by jamesmikesell** [comment]
@rhatdan I don't have any modifications to the podman code base, and as such can't do a PR.  My notes above were just commands that had to be manually executed against a running podman machine instance.

**Comment by rhatdan** [comment]
Those commands would be useful to add potentially to a ignition script potentially.

You don't need modifications rights to the podman code, anyone can open a PR.

**Comment by baude** [comment]
fcos offers a zram swap mechanism.  would that be sufficient for people?
